### PR TITLE
add demos and details for aida code gen

### PR DESCRIPTION
**Pull request by abhinavrastogi-harness**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: docs/platform/harness-aida/aida-code-gen.md [hunk 1]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+
+### Model-based Solution for Optimal Prompt Finding (WIP)
+
+At the heart of Harness AIDA Code Assistant's effectiveness is its model-based approach for finding the optimal prompt for the LLM. By carefully crafting prompts based on the collected context and static analysis, Harness AIDA Code Assistant ensures that the suggestions provided are both relevant and accurate. This approach maximizes the utility of the LLM, enabling it to generate high-quality code completions and other development aids.
+This technical framework sets Harness AIDA Code Assistant apart as a comprehensive tool that addresses the multifaceted challenges of software development. By leveraging AI in these innovative ways, Harness AIDA Code Assistant not only enhances developer productivity but also contributes to the creation of higher quality software.
 
-* VSCode (>= 1.86.0)
+## Usage and Implementation of Harness AIDA Code Assistant
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:42]:
```suggestion
## Use and implement Harness AIDA Code Assistant
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 2]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
 
-## Installation
+Integrating Harness AIDA Code Assistant into the software development workflow is a straightforward process designed to minimize setup time and maximize efficiency. This section outlines the steps to implement Harness AIDA Code Assistant and demonstrates its practical use in everyday coding tasks.
 
-It is very easy to get started with Harness AIDA.
+### Setting Up the IDE Extension
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:46]:
```suggestion
### Set up the IDE extension
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 3]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
 
-1. Download the official extension from the marketplace of any supported IDE.
-2. [Generate a Harness Personal Access Token.](/docs/platform/automation/api/add-and-manage-api-keys)
-3. Add the token in the extension settings.
+The extension is currently only available to our alpha partners. To begin using Harness AIDA Code Assistant, developers first install the extension from their IDE's marketplace. This process is similar to adding any other IDE plugin, ensuring ease of adoption. Once installed, developers can configure the extension settings to tailor Harness AIDA Code Assistant's behavior to their preferences, such as specifying which programming languages to activate it for or adjusting the sensitivity of its suggestions.
 
-## Capabilities
+### Securing API Access
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:50]:
```suggestion
### Secure API access
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 4]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
 
-### Code Generation
+Harness AIDA Code Assistant requires access to its AI model through Harness Gateway. This token ensures secure communication between the IDE extension and Harness AIDA Code Assistant's servers, safeguarding the code and the developer's intellectual property. The setup guide provides detailed instructions on obtaining and configuring the API token within the IDE.
 
-tbd
+### Real-world Coding Scenarios
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:54]:
```suggestion
### Real world coding scenarios
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 5]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+
+With Harness AIDA Code Assistant activated, developers can start leveraging its features to streamline their coding tasks. For example, when writing new functions, Harness AIDA Code Assistant can suggest complete blocks of code that adhere to best practices, significantly reducing the time required to write boilerplate code. In debugging sessions, Harness AIDA Code Assistant can suggest potential fixes or even generate unit tests to cover edge cases that the developer might not have considered.
+
+## Features
+
+### Code Autocomplete
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:60]: To minimize impact of the size of videos on an already oversized repo, we really need to get these videos on YouTube ASAP and not rely on long-term storage in HDH's repo.

File: docs/platform/harness-aida/aida-code-gen.md [hunk 6]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+<DocVideo src="./static/test_generation_demo.mov" />
+
+## Supported IDEs
+
+Currently we support the following IDEs:
+
+- VSCode (>= 1.86.0)
 
-### Comment Generation / Explain Code
+## Ongoing Improvements and Research Directions
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:84]:
```suggestion
## Ongoing improvements and research directions
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 7]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
 
-tbd
+Harness AIDA Code Assistant is in a constant state of evolution, with ongoing efforts to expand its capabilities and improve its performance. Some areas of focus include:
 
-### Generate Terraform configurations
--->
\ No newline at end of file
+- Enhancing Contextual Awareness: By refining the models' understanding of code context, Harness AIDA Code Assistant aims to provide even more relevant and accurate suggestions.
+- Support for More Languages and Frameworks: Expanding the range of programming languages and development frameworks that Harness AIDA Code Assistant supports is a priority, making it a more versatile tool for a broader audience.
+- User Interface Improvements: Based on user feedback, the Harness AIDA Code Assistant team continually works on making the tool more intuitive and user-friendly.
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:90]:
```suggestion
- User interface improvements: Based on user feedback, the Harness AIDA Code Assistant team continually works on making the tool more intuitive and user-friendly.
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 8]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+## Ongoing Improvements and Research Directions
 
-tbd
+Harness AIDA Code Assistant is in a constant state of evolution, with ongoing efforts to expand its capabilities and improve its performance. Some areas of focus include:
 
-### Generate Terraform configurations
--->
\ No newline at end of file
+- Enhancing Contextual Awareness: By refining the models' understanding of code context, Harness AIDA Code Assistant aims to provide even more relevant and accurate suggestions.
+- Support for More Languages and Frameworks: Expanding the range of programming languages and development frameworks that Harness AIDA Code Assistant supports is a priority, making it a more versatile tool for a broader audience.
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:89]:
```suggestion
- Support for more languages and frameworks: Expanding the range of programming languages and development frameworks that Harness AIDA Code Assistant supports is a priority, making it a more versatile tool for a broader audience.
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 9]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
-### Comment Generation / Explain Code
+## Ongoing Improvements and Research Directions
 
-tbd
+Harness AIDA Code Assistant is in a constant state of evolution, with ongoing efforts to expand its capabilities and improve its performance. Some areas of focus include:
 
-### Generate Terraform configurations
--->
\ No newline at end of file
+- Enhancing Contextual Awareness: By refining the models' understanding of code context, Harness AIDA Code Assistant aims to provide even more relevant and accurate suggestions.
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:88]:
```suggestion
- Enhancing contextual awareness: By refining the models' understanding of code context, Harness AIDA Code Assistant aims to provide even more relevant and accurate suggestions.
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 10]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
 Code Generation with AIDA is an IDE plugin. It is currently in development. Check back soon for more details.
 
-<!--
+## Technical Framework of Harness AIDA Code Assistant
 
-## Supported IDEs
+At the core of Harness AIDA Code Assistant's functionality is its sophisticated use of Large Language Models (LLMs), which are trained on vast datasets of code to understand and generate human-like code. These LLMs include but are not limited to OpenAI GPT models like GPT 3.5turbo and GPT 4.0. This allows Harness AIDA Code Assistant to offer contextually relevant code suggestions, explanations, and test generation.
 
-Currently we support the following IDEs:
+### Integrated Development Environment (IDE) Extension
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:21]:
```suggestion
### Integrated Development Environment (IDE) extension
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 11]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+
+Harness AIDA Code Assistant functions as an extension for popular Integrated Development Environments (IDEs). This design choice ensures that developers can use Harness AIDA Code Assistant's features without leaving their familiar coding environment. The extension is built to be lightweight and responsive, providing suggestions in real-time as developers type. It supports a wide range of programming languages, making it versatile across different development contexts.
+
+### Agent-based Context Collection
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:25]:
```suggestion
### Agent-based context collection
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 12]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+
+To provide relevant suggestions, Harness AIDA Code Assistant employs an agent-based approach for context collection. This involves analyzing the current codebase, open files, and cursor position to understand the developer's intent and the task at hand. By gathering this context, Harness AIDA Code Assistant can tailor its suggestions to fit the specific requirements of the code being written.
+
+### Enriched Context through Static Code Analysis
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:29]:
```suggestion
### Enriched context through static code analysis
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 13]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+
+Further enhancing its context-aware capabilities, Harness AIDA Code Assistant incorporates static code analysis to understand the structure and semantics of the code. This analysis allows Harness AIDA Code Assistant to identify contextually relevant code that may live far from the file that is currently being edited
+
+### External Documentation Fetching (WIP)
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:33]:
```suggestion
### External documentation fetching (WIP)
```

File: docs/platform/harness-aida/aida-code-gen.md [hunk 14]
```diff
@@ -14,35 +14,78 @@ Whether you're a seasoned developer looking to speed up your workflow or a begin
...
+
+Recognizing the importance of accurate documentation, Harness AIDA Code Assistant includes a feature to fetch relevant external documentation. Whether it's official language documentation, framework guides, or community-contributed notes, Harness AIDA Code Assistant can present this information directly within the IDE. This saves developers time and ensures that they have access to the information they need when they need it.
+
+### Model-based Solution for Optimal Prompt Finding (WIP)
```
Comment (aimurphy) [inline docs/platform/harness-aida/aida-code-gen.md:37]:
```suggestion
### Model-based solution for optimal prompt finding (WIP)
```

**Review by aimurphy — CHANGES_REQUESTED** [review]
Per our style guide, please use sentence case for headings. 

Please note my comment about the videos for the long term. 

Otherwise, lgtm.
